File: Solutions/Python/longest common prefix.py
# Problem link https://practice.geeksforgeeks.org/problems/longest-common-prefix-in-an-array5129/1

def longestCommonPrefix( a):
	
	size = len(a)

	if (size == 0):
		return ""

	if (size == 1):
		return a[0]

	a.sort()
	
	end = min(len(a[0]), len(a[size - 1]))

	i = 0
	while (i < end and
		a[0][i] == a[size - 1][i]):
		i += 1

	pre = a[0][0: i]
	return pre

# Driver Code
if __name__ == "__main__":

	input = ["flower", "flow",
					"flight", "flown"]
	print("The longest Common Prefix is :" ,
				longestCommonPrefix(input))

File: Solutions/C++/CPP/Longest Common Prefix.cpp
// Problem link https://practice.geeksforgeeks.org/problems/longest-common-prefix-in-an-array5129/1


class Solution{
  public:
  
    string LCPUtil(string &str1, string &str2)
    {
        string result = "";
        int len = min(str1.length(), str2.length());
        
        for(int i = 0; i < len; i++)
        {
            if(str1[i] != str2[i])
                break;
            
            result += str1[i];
        }
        
        return result;

    }
    
    string longestCommonPrefix (string arr[], int N)
    {
        // your code here
        
        string prefix = arr[0];
        
        for(int i = 1; i < N; i++)
            prefix = LCPUtil(prefix, arr[i]);
            
        return prefix;
    }
};